libarrow-cpp missing some modules [Enhancement]:

### Problem description

The package `libarrow-cpp` is missing some `.so` modules which causes error while installing python package `pyarrow`.
_
Full logs: https://pastebin.com/nxspDNcE

 Error
  ```error
-- Found Arrow: /data/data/com.termux/files/usr/include (found version "7.0.0")
      -- Arrow version: 7.0.0 (CMake package configuration: Arrow)
      -- Arrow SO and ABI version: 700
      -- Arrow full SO version: 700.0.0
      -- Found the Arrow core shared library: /data/data/com.termux/files/usr/lib/libarrow.so
      -- Found the Arrow core import library:
      -- Found the Arrow core static library: /data/data/com.termux/files/usr/lib/libarrow.a
      -- Could NOT find ArrowPython (missing: ArrowPython_DIR)
      -- Checking for module 'arrow-python'
      --   No package 'arrow-python' found
      CMake Error at /data/data/com.termux/files/usr/share/cmake-3.23/Modules/FindPackageHandleStandardArgs.cmake:230 (message):
        Could NOT find ArrowPython (missing: ARROW_PYTHON_INCLUDE_DIR
        ARROW_PYTHON_LIB_DIR) (found version "7.0.0")
      Call Stack (most recent call first):
        /data/data/com.termux/files/usr/share/cmake-3.23/Modules/FindPackageHandleStandardArgs.cmake:594 (_FPHSA_FAILURE_MESSAGE)
```
__
### What steps will reproduce the bug?

``pkg install libarrow-cpp libarrow-cpp-static``
``pip install pyarrow``


### What is the expected behavior?

The package should have installed properly.

### System information

termux-info:

```
Termux Variables:
TERMUX_API_VERSION=0.50.1
TERMUX_APK_RELEASE=GITHUB
TERMUX_APP_PID=8447
TERMUX_IS_DEBUGGABLE_BUILD=1
TERMUX_MAIN_PACKAGE_FORMAT=debian
TERMUX_VERSION=0.118.0
Packages CPU architecture:
aarch64
Subscribed repositories:
# sources.list
deb https://termux.mentality.rip/termux-main/ stable main
# sources.list.d/pointless.list
deb https://its-pointless.github.io/files/21 termux extras
Updatable packages:
gnupg/stable 2.3.5 aarch64 [upgradable from: 2.3.4]
gpgv/stable 2.3.5 aarch64 [upgradable from: 2.3.4]
zstd/stable 1.5.2-1 aarch64 [upgradable from: 1.5.2]
Android version:
11
Kernel build information:
Linux localhost 4.14.186+ #1 SMP PREEMPT Thu Feb 10 17:02:37 CST 2022 aarch64 Android
Device manufacturer:
realme
Device model:
RMX2151

```

--- a/packages/libarrow-cpp/build.sh
+++ b/packages/libarrow-cpp/build.sh
@@ -2,15 +2,48 @@ TERMUX_PKG_HOMEPAGE=https://github.com/apache/arrow
 TERMUX_PKG_DESCRIPTION="C++ libraries for Apache Arrow"
 TERMUX_PKG_LICENSE="Apache-2.0"
 TERMUX_PKG_MAINTAINER="@termux"
-TERMUX_PKG_VERSION=7.0.0
-TERMUX_PKG_SRCURL=https://github.com/apache/arrow/archive/refs/tags/apache-arrow-${TERMUX_PKG_VERSION}.tar.gz
-TERMUX_PKG_SHA256=57e13c62f27b710e1de54fd30faed612aefa22aa41fa2c0c3bacd204dd18a8f3
-TERMUX_PKG_DEPENDS="libc++"
+TERMUX_PKG_VERSION=(7.0.0)
+TERMUX_PKG_REVISION=1
+TERMUX_PKG_VERSION+=(1.22.3) # NumPy version
+TERMUX_PKG_SRCURL=(https://github.com/apache/arrow/archive/refs/tags/apache-arrow-${TERMUX_PKG_VERSION}.tar.gz
+                   https://github.com/numpy/numpy/archive/refs/tags/v${TERMUX_PKG_VERSION[1]}.tar.gz)
+TERMUX_PKG_SHA256=(57e13c62f27b710e1de54fd30faed612aefa22aa41fa2c0c3bacd204dd18a8f3
+                   c8f3ec591e3f17b939220f2b9eabb4c5e2db330f8af62c0a3aeee8a4d1a6c0db)
+TERMUX_PKG_DEPENDS="libc++, libre2, utf8proc"
+TERMUX_PKG_BUILD_DEPENDS="rapidjson"
 TERMUX_PKG_EXTRA_CONFIGURE_ARGS="
+-DARROW_BUILD_STATIC=OFF
 -DARROW_JEMALLOC=OFF
+-DARROW_PYTHON=ON
 "
 
+termux_step_post_get_source() {
+	mv numpy-${TERMUX_PKG_VERSION[1]} numpy
+}
+
 termux_step_pre_configure() {
+	_PYTHON_VERSION=$(. $TERMUX_SCRIPTDIR/packages/python/build.sh; echo $_MAJOR_VERSION)
+	termux_setup_python_crossenv
+	pushd $TERMUX_PYTHON_CROSSENV_SRCDIR
+	_CROSSENV_PREFIX=$TERMUX_PKG_BUILDDIR/python-crossenv-prefix
+	python${_PYTHON_VERSION} -m crossenv \
+		$TERMUX_PREFIX/bin/python${_PYTHON_VERSION} \
+		${_CROSSENV_PREFIX}
+	popd
+	. ${_CROSSENV_PREFIX}/bin/activate
+	LDFLAGS+=" -lpython${_PYTHON_VERSION}"
+	export NPY_DISABLE_SVML=1
+	pushd $TERMUX_PKG_SRCDIR/numpy
+	pip install .
+	popd
+	TERMUX_PKG_EXTRA_CONFIGURE_ARGS+="
+		-DPYTHON_EXECUTABLE=python
+		-DPYTHON_INCLUDE_DIRS=$TERMUX_PREFIX/include/python${_PYTHON_VERSION}
+		-DPYTHON_LIBRARIES=$TERMUX_PREFIX/lib/libpython${_PYTHON_VERSION}.so
+		-DPYTHON_OTHER_LIBS=
+		-DNUMPY_INCLUDE_DIRS=${_CROSSENV_PREFIX}/cross/lib/python${_PYTHON_VERSION}/site-packages/numpy/core/include
+		"
+
 	TERMUX_PKG_SRCDIR+="/cpp"
 
 	_NEED_DUMMY_LIBRT_A=

--- a/packages/libarrow-cpp/cpp-cmake_modules-FindPython3Alt.cmake.patch
+++ b/packages/libarrow-cpp/cpp-cmake_modules-FindPython3Alt.cmake.patch
@@ -1,0 +1,18 @@
+--- a/cpp/cmake_modules/FindPython3Alt.cmake
++++ b/cpp/cmake_modules/FindPython3Alt.cmake
+@@ -23,6 +23,7 @@
+ # - PYTHON_OTHER_LIBS
+ # - NUMPY_INCLUDE_DIRS
+ 
++if(FALSE)
+ # Need CMake 3.15 or later for Python3_FIND_STRATEGY
+ if(${CMAKE_VERSION} VERSION_LESS "3.15.0")
+   # Use deprecated Python- and NumPy-finding code
+@@ -69,6 +70,7 @@
+ set(PYTHON_OTHER_LIBS)
+ 
+ get_target_property(NUMPY_INCLUDE_DIRS Python3::NumPy INTERFACE_INCLUDE_DIRECTORIES)
++endif(FALSE)
+ 
+ # CMake's python3_add_library() doesn't apply the required extension suffix,
+ # detect it ourselves.

--- a/packages/libarrow-cpp/libarrow-python.subpackage.sh
+++ b/packages/libarrow-cpp/libarrow-python.subpackage.sh
@@ -1,0 +1,8 @@
+TERMUX_SUBPKG_INCLUDE="
+lib/libarrow_python*
+lib/cmake/arrow/*ArrowPython*.cmake
+lib/pkgconfig/arrow-python.pc
+include/arrow/python
+"
+TERMUX_SUBPKG_DESCRIPTION="Arrow CPython extensions"
+TERMUX_SUBPKG_DEPENDS="python"

